src/python_testing/TC_CADMIN_1_11.py

```diff
@@ -0,0 +1,210 @@
+        while True:
+            random_value = random.randint(10000000, 99999999)
+            if random_value != value:
+                return random_value
+
+    async def CommissionAttempt(
+            self, setupPinCode: int):
+
+        logging.info("-----------------Commissioning with TH_CR2-------------------------")
+        await self.th2.CommissionOnNetwork(
+            nodeId=self.dut_node_id, setupPinCode=setupPinCode,
+            filterType=ChipDeviceCtrl.DiscoveryFilterType.LONG_DISCRIMINATOR, filter=self.discriminator)
+
+    def pics_TC_CADMIN_1_11(self) -> list[str]:
+        return ["CADMIN.S"]
+
+    @async_test_body
+    async def test_TC_CADMIN_1_11(self):
+        self.step(1)
+
+        # Establishing TH1 and TH2 controllers
+        self.th1 = self.default_controller
+        self.discriminator = random.randint(0, 4095)
+        th2_certificate_authority = self.certificate_authority_manager.NewCertificateAuthority()
+        th2_fabric_admin = th2_certificate_authority.NewFabricAdmin(vendorId=0xFFF1, fabricId=self.th1.fabricId + 1)
+        self.th2 = th2_fabric_admin.NewController(nodeId=2, useTestCommissioner=True)
+
+        self.step(2)
+        GC_cluster = Clusters.GeneralCommissioning
+        attribute = GC_cluster.Attributes.BasicCommissioningInfo
+        duration = await self.read_single_attribute_check_success(endpoint=0, cluster=GC_cluster, attribute=attribute)
+        self.timeout = duration.maxCumulativeFailsafeSeconds
+
+        self.step(3)
+        params = await self.OpenCommissioningWindow(self.th1, 0x00)
+        setupPinCode = params.setupPinCode
+        busy_enum = Clusters.AdministratorCommissioning.Enums.StatusCode.kBusy
+
+        self.step(4)
+        await self.CommissionAttempt(setupPinCode)
+
+        self.step(5)
+        await self.OpenCommissioningWindow(self.th1, 0x00)
+
+        self.step(6)
+        await self.OpenCommissioningWindow(self.th1, busy_enum)
+
+        self.step(7)
+        await self.OpenCommissioningWindow(self.th2, busy_enum)
+
+        self.step(8)
+        revokeCmd = Clusters.AdministratorCommissioning.Commands.RevokeCommissioning()
+        await self.th1.SendCommand(nodeid=self.dut_node_id, endpoint=0, payload=revokeCmd, timedRequestTimeoutMs=6000)
+        # The failsafe cleanup is scheduled after the command completes, so give it a bit of time to do that
+        sleep(1)
+
+        self.step(9)
+
+        AC_cluster = Clusters.AdministratorCommissioning
+        if Clusters.Objects.AdministratorCommissioning.featureMap is not None:
+            fm_attribute = Clusters.AdministratorCommissioning.Attributes
+            features = await self.read_single_attribute_check_success(cluster=AC_cluster, attribute=fm_attribute.FeatureMap)
+        else:
+            features = AC_cluster.Attributes.FeatureMap.value
+
+        self.supports_bc = bool(features & AC_cluster.Bitmaps.Feature.kBasic) != 0
+
+        if self.supports_bc:
```

Comment:
Do you know if the current example app supports BC? I want to ensure the test steps have all been checked by the CI / you testing.

Reply:
Hi Cecille,
After your provided guidance and support on how to use the zap tool, we were able to enable BC feature on the all-clusters app then validated that the updated method provided in the latest changes did work to open the BC commissioning window.

Reply:
kk, sounds like this was tested, so we're good here.